docs/examples/platform/posix/posix_single_threaded_executor.hpp

```diff
@@ -87,13 +87,13 @@ class PosixSingleThreadedExecutor final : public libcyphal::platform::SingleThre
         // so we can reuse them, and grow on demand (but never shrink).
         //
         poll_fds_.clear();
```

Comment:
I think it's worth adding a comment (considering that this is an example) that one could possibly avoid the need to traverse the awaitables on every poll by switching to another multiplexing API that stores fds in a kernel space structure, like epoll. Each awaitable item would then register and unregister itself with that kernel space object as needed and the poll would go straight to the blocking call without the need to reconstruct anything.

Reply:
I would like (a bit later, separate dedicated PR) to do such exercise - switch to `::epoll`.